frontend/src/v5/ui/components/shared/modalsDispatcher/templates/imagesModal/imagesModal.component.tsx

```diff
@@ -49,35 +63,117 @@ export const ImagesModal = ({ images, displayImageIndex = 0, onClickClose, open
 				changeImageIndex(1);
 				break;
 			default:
-				// do nothing
+			// do nothing
 		}
 	};
 
+	const handleDelete = () => {
+		onDelete(imageIndex);
+		if (imageIndex === (imagesLength - 1)) {
+			setImageIndex(imageIndex - 1);
+		}
+	};
+
+	const triggerDeleteModal = () => {
+		DialogsActionsDispatchers.open('delete', {
+			onClickConfirm: handleDelete,
+			titleLabel: formatMessage({
+				id: 'imagesModal.deleteModal.title',
+				defaultMessage: 'Delete image',
+			}),
+			message: formatMessage({
+				id: 'imagesModal.deleteModal.message',
+				defaultMessage: 'Are you sure you want to delete the selected image?',
+			}),
+		});
+	};
+
+	const centerSelectedThumbnail = () => {
+		if (!imageRef.current) return;
+		// @ts-ignore
+		imageRef.current.scrollIntoView({ behavior: 'instant', inline: 'center' });
+	};
+
 	useEffect(() => {
 		document.addEventListener('keydown', handleKeyDown);
 		return () => document.removeEventListener('keydown', handleKeyDown);
-	});
+	}, []);
+
+	useEffect(() => {
+		if (!imagesLength) {
+			onClickClose();
+		}
+	}, [imagesLength]);
+
+	useEffect(() => { centerSelectedThumbnail(); }, [imagesLength, imageIndex]);
 
 	return (
 		<Modal open={open} onClose={onClickClose}>
-			<Container>
-				{imagesLength === 1 ? (
-					<Image src={images[imageIndex]} />
-				) : (
-					<>
-						<PreviousButton onClick={() => changeImageIndex(-1)}>
+			<TopBar>
+				<Buttons>
+					{hasManyImages && (
+						<Counter $counterChars={imagesLength.toString().length}>
+							<span>{imageIndex + 1}</span>
```

Comment:
I was doing that at the beginning, but to style the content, each element was to be wrapped with some component (in this case, I used the `span`).

Using a formattedMessage where I pass the values, would look like this
```ts
<FormattedMessage
	id="images.count.of"
	defaultMessage="{current} {of} {total}"
	values={{
		current: <span>{imageIndex + 1}</span>,
		of: <span><FormattedMessage id="images.count.of" defaultMessage="of" /></span>,
		total: <span>{imagesLength}</span>,
	}}
/>
```
which is pretty much what I did. If you think this is cleaner or it adds something that I am missing, I'm happy to push the change

Reply:
I did that because of a visual oddity: when you change image index, the width of "count" changes affecting the overall width of the counter component; and since that is the first element of the row, the other buttons are shifted left and right everytime the count changes.

I have, however, found a fix using css, so I will fix this